### ISSUE TITLE
PVR Client not updated on ubuntu EOAN repository

### ISSUE THREAD

**Issue by gd71** [aassif/pvr.freebox#43]
Hi there,
You're probably already aware but in doubt.
Installation fail with "Addon incompatible with kodi"
My guess is regarding the requirement from addon.xml
```
<import addon="kodi.binary.instance.pvr" minversion="6.1.0" version="6.1.0"/>
```
compared to what is installed on my ubuntu 19.10
```
<addon id="kodi.binary.instance.pvr" version="6.3.0" provider-name="Team Kodi">
```
But I may be wrong
Thanks for your helpfull and really efficient addon!
PS: BTW, librelec binary for Leïa works well on kodi Leïa in a raspberry py 4b raspbian os based ;)

**Comment by gd71** [comment]
Looks like ubuntu eoan repo is not updated for amd64 target 
http://ppa.launchpad.net/team-xbmc/xbmc-nightly/ubuntu/pool/main/k/kodi-pvr-freebox/

**Comment by gd71** [comment]
I've temporarily used the one provided by the disco repository (v2.2.3.1) and it works well.
Nevertheless I updates must be made by hand and no one knows if a dependency might be broken in the future.
I suggest keeping this issue open unless you have a reason for not publishing the plugin on last ubuntu release

**Comment by aassif** [comment]
Indeed, the last build failed because of missing #defines:
https://launchpad.net/~team-xbmc/+archive/ubuntu/xbmc-nightly/+build/18899910

However, I don't know why they're missing as the Kodi PR is already merged:
https://github.com/xbmc/xbmc/pull/17496

**Comment by aassif** [comment]
@wsnipex What should I do?

**Comment by aassif** [comment]
In fact, the same problem occurs with other PVR clients:
https://launchpad.net/~team-xbmc/+archive/ubuntu/xbmc-nightly/+packages?field.name_filter=kodi-pvr&field.status_filter=published&field.series_filter=

**Comment by gd71** [comment]
Lots of stuff currently on PVR implementation.
didn't see that was also the case for other clients.
Looks like this issue is not on your side, therefore, I don't know who could take care of nor who is in charge of maintaining ubuntu releases
XBMC project?

**Comment by gd71** [comment]
Received yesterday an update for kodi-pvr-iptvsimple - 4.13.1-1~eoan amd64
Hopefully next release will build correctly

**Comment by gd71** [comment]
Hello,
the lest release 2.2.5 has been successfully updated on eoan x86_64 repo this morning.
Unfortunately, the addon is not compatible with the last nightly 20200425. :)

**Comment by gd71** [comment]
Indeed PVR Api is now 6.4.0

**Comment by aassif** [comment]
Strange.
I updated everyhting today and I have no issue.
Focal Fossa, Kodi 19.0-ALPHA1 Git:20200424-c4f5c5e4d1.

**Comment by gd71** [comment]
Received on ubuntu 
```
$ apt-show-versions kodi
kodi:all/eoan 2:19.0+git20200425.0314-35641bab4e-0eoan uptodate
```
Please note the update is 20200425 not 20200424

I've rolled back since and now it's working but I guess you will have the issue soon also

**Comment by gd71** [comment]
https://github.com/xbmc/xbmc/commit/f7416ca0e3978abf15129a29cac9d6582e9c5aa3
https://github.com/xbmc/xbmc/pull/17586

**Comment by aassif** [comment]
Indeed, I now have the same issue.
Yet, the [6.4.0 PVR API](https://github.com/aassif/pvr.freebox/pull/41) was merged yesterday.

**Comment by gd71** [comment]
could it be that this build has been made before 6.4.0 API merged into xbmc master branch?
What is the build status for other platforms ?

**Comment by aassif** [comment]
https://jenkins.kodi.tv/view/Addons/job/aassif/job/pvr.freebox/job/Matrix/69/

**Comment by aassif** [comment]
6.4.0 for everyone.

**Comment by gd71** [comment]
just removed completely and reinstalled.
```
/usr/share/kodi/addons$ls -ltr pvr.freebox/
total 16
-rw-r--r-- 1 root root 5218 juin   1  2013 icon.png
-rw-r--r-- 1 root root 1266 juin   1  2013 addon.xml
drwxr-xr-x 3 root root 4096 avril 26 18:19 resources
/usr/share/kodi/addons$ cat pvr.freebox/addon.xml 
<?xml version="1.0" encoding="UTF-8"?>
<addon
  id="pvr.freebox"
  version="2.2.5"
  name="PVR Freebox TV"
  provider-name="aassif">
  <requires>
<import addon="kodi.binary.global.main" minversion="1.0.12" version="1.1.0"/>
<import addon="kodi.binary.global.gui" minversion="5.14.0" version="5.14.0"/>
<import addon="kodi.binary.instance.pvr" minversion="6.3.0" version="6.3.0"/></requires>
  <extension
    point="xbmc.pvrclient"
    library_linux="pvr.freebox.so.2.2.5"/>
  <extension point="xbmc.addon.metadata">
    <summary lang="fr_FR">Extension enregistreur vidéo (PVR) pour Freebox TV. https://github.com/aassif/pvr.freebox</summary>
    <summary lang="en_GB">Freebox TV PVR. https://github.com/aassif/pvr.freebox</summary>
    <description lang="fr_FR">Le client enregistreur vidéo (PVR) gère les chaînes Freebox TV, le guide électronique des programmes et les enregistrements.</description>
    <description lang="en_GB">This PVR client handles Freebox TV channels, EPG and recordings.</description>
    <disclaimer lang="fr_FR">Logiciel en cours d'élaboration !</disclaimer>
    <disclaimer lang="en_GB">This is unstable software!</disclaimer>
    <platform></platform>
    <assets>
      <icon>icon.png</icon>
    </assets>
  </extension>
</addon>
```
I'm not used with binary addons but the addon.xml file shiped with the .deb package delivered yesterday looks outdated.
could it be the issue?

**Comment by gd71** [comment]
```
/usr/share/kodi/addons$ dpkg-query -L kodi-pvr-freebox
/.
/usr
/usr/lib
/usr/lib/x86_64-linux-gnu
/usr/lib/x86_64-linux-gnu/kodi
/usr/lib/x86_64-linux-gnu/kodi/addons
/usr/lib/x86_64-linux-gnu/kodi/addons/pvr.freebox
/usr/lib/x86_64-linux-gnu/kodi/addons/pvr.freebox/pvr.freebox.so.2.2.5
/usr/share
/usr/share/doc
/usr/share/doc/kodi-pvr-freebox
/usr/share/doc/kodi-pvr-freebox/changelog.Debian.gz
/usr/share/doc/kodi-pvr-freebox/copyright
/usr/share/kodi
/usr/share/kodi/addons
/usr/share/kodi/addons/pvr.freebox
/usr/share/kodi/addons/pvr.freebox/addon.xml
/usr/share/kodi/addons/pvr.freebox/icon.png
/usr/share/kodi/addons/pvr.freebox/resources
/usr/share/kodi/addons/pvr.freebox/resources/language
/usr/share/kodi/addons/pvr.freebox/resources/language/resource.language.en_gb
/usr/share/kodi/addons/pvr.freebox/resources/language/resource.language.en_gb/strings.po
/usr/share/kodi/addons/pvr.freebox/resources/language/resource.language.fr_fr
/usr/share/kodi/addons/pvr.freebox/resources/language/resource.language.fr_fr/strings.po
/usr/share/kodi/addons/pvr.freebox/resources/settings.xml
/usr/lib/x86_64-linux-gnu/kodi/addons/pvr.freebox/pvr.freebox.so
/usr/lib/x86_64-linux-gnu/kodi/addons/pvr.freebox/pvr.freebox.so.19.0
```

**Comment by gd71** [comment]
looks like all ubuntu-ppa deliveries have the same issue

**Comment by gd71** [comment]
Looks like this one is promising :)
https://github.com/xbmc/xbmc/pull/17775

**Comment by aassif** [comment]
Not really.
The only problem is that Kodi is not up-to-date on the build system.
Hence the different PVR API version.
Things should sort themselves out. Just be patient.

**Comment by gd71** [comment]
fixed with 2.2.6 release
